src/exporter/DaemonMetricCollector.cc

```diff
@@ -353,14 +359,6 @@ void DaemonMetricCollector::dump_asok_metric(json_object perf_info,
     json_value sum_value = perf_values.as_object()["sum"];
     add_double_or_int_metric(builder, sum_value, name + "_sum", description,
                              metric_type, labels);
-  } else if (type & PERFCOUNTER_TIME) {
```

Comment:
It looks like the `PERFCOUNTER_LONGRUNAVG` branch above has a defect too.  Why is the `_count` metric has the same type as the `_sum` metric?  Shouldn't the `_count` metric be a counter unconditionally?
```
# HELP ceph_rbd_mirror_snapshot_sync_time_count Average sync time
# TYPE ceph_rbd_mirror_snapshot_sync_time_count gauge
                                                ^^^^^
ceph_rbd_mirror_snapshot_sync_time_count{ceph_daemon="client.admin.689771"} 2
```

Reply:
> It looks like the `PERFCOUNTER_LONGRUNAVG` branch above has a defect too. Why is the `_count` metric has the same type as the `_sum` metric? Shouldn't the `_count` metric be a counter unconditionally?
> 
> ```
> # HELP ceph_rbd_mirror_snapshot_sync_time_count Average sync time
> # TYPE ceph_rbd_mirror_snapshot_sync_time_count gauge
>                                                 ^^^^^
> ceph_rbd_mirror_snapshot_sync_time_count{ceph_daemon="client.admin.689771"} 2
> ```

Not sure, but  it's the same what prometheus module has
https://github.com/ceph/ceph/blob/main/src/pybind/mgr/prometheus/module.py#L1651

Reply:
> Not sure, but it's the same what prometheus module has https://github.com/ceph/ceph/blob/main/src/pybind/mgr/prometheus/module.py#L1651

No, it's not the same.  Looking at exactly the `if` block that you linked:
https://github.com/ceph/ceph/blob/b02e1b41f1dcd0aa36e0ce329a2c567a1a30bab5/src/pybind/mgr/prometheus/module.py#L1651-L1668
On line 1655 `stattype` variable is passed, whereas on line 1664 `'counter'` literal is passed unconditionally.

Reply:
> > Not sure, but it's the same what prometheus module has https://github.com/ceph/ceph/blob/main/src/pybind/mgr/prometheus/module.py#L1651
> 
> No, it's not the same. Looking at exactly the `if` block that you linked:
> 
> https://github.com/ceph/ceph/blob/b02e1b41f1dcd0aa36e0ce329a2c567a1a30bab5/src/pybind/mgr/prometheus/module.py#L1651-L1668
> 
> On line 1655 `stattype` variable is passed, whereas on line 1664 `'counter'` literal is passed unconditionally.

Sorry I'm not following exactly, from what I understand we should keep the metric type same which is coming from schema output and that's the case here as well, the type is gauge so we exposed it as gauge only.

Reply:
https://github.com/ceph/ceph/blob/main/doc/dev/perf_counters.rst?plain=1#L66

Reply:
Both prometheus module and exporter daemon split `PERFCOUNTER_LONGRUNAVG` perf counter into two metrics:  `<perf counter name>_sum` and `<perf counter name>_count`.  The former is a sum of values, so it makes sense to preserve the metric type.  The latter is a count of values, so its type should be `counter`.  Prometheus module gets this right:
https://github.com/ceph/ceph/blob/b02e1b41f1dcd0aa36e0ce329a2c567a1a30bab5/src/pybind/mgr/prometheus/module.py#L1664

Reply:
Yeah I did check that by verifying the /metrics output from module too and also description needed to be adopted for counters and sum types of metrics which I've now fixed.

Reply:
Please post the outputs again, but this time do two images with an additional snapshot for each.  After mirroring is enabled, do something like:
```
$ rbd bench --io-threads 1 --io-total 20M --io-type write --io-pattern rand <poolname>/<imagename>
$ rbd mirror image snapshot <poolname>/<imagename>
```
and wait for a minute before grabbing the outputs.

Reply:
> Please post the outputs again, but this time do two images with an additional snapshot for each. After mirroring is enabled, do something like:
> 
> ```
> $ rbd bench --io-threads 1 --io-total 20M --io-type write --io-pattern rand <poolname>/<imagename>
> $ rbd mirror image snapshot <poolname>/<imagename>
> ```
> 
> and wait for a minute before grabbing the outputs.

Sure. I'll need to build again, will paste the output once I've build ready.